IncrementalSVDBasisGenerator.C

```diff
@@ -18,43 +18,28 @@
 namespace CAROM {
 
 IncrementalSVDBasisGenerator::IncrementalSVDBasisGenerator(
-   int dim,
-   double linearity_tol,
-   bool skip_linearly_dependent,
-   bool fast_update,
-   int max_basis_dimension,
-   double initial_dt,
-   int samples_per_time_interval,
-   double sampling_tol,
-   double max_time_between_samples,
+   IncrementalSVDBasisGeneratorOptions options,
    const std::string& basis_file_name,
-   bool save_state,
-   bool restore_state,
-   bool updateRightSV,
-   Database::formats file_format,
-   double min_sampling_time_step_scale,
-   double sampling_time_step_scale,
-   double max_sampling_time_step_scale,
-   bool debug_algorithm) :
+   Database::formats file_format) :
    SVDBasisGenerator(basis_file_name, file_format)
 {
-   d_svdsampler.reset(new IncrementalSVDSampler(dim,
-                                                linearity_tol,
-                                                skip_linearly_dependent,
-                                                fast_update,
-                                                max_basis_dimension,
-                                                initial_dt,
-                                                samples_per_time_interval,
-                                                sampling_tol,
-                                                max_time_between_samples,
+   d_svdsampler.reset(new IncrementalSVDSampler(options.dim,
```

Comment:
There may be many classes that have many constructor arguments, and `IncrementalSVDSampler` is just one, but what do you think about simplifying its constructor as well by passing `options`? If `IncrementalSVDSampler` is an internal class not called by users typically, then it is not so bad, but the libROM code would still be cleaner with a struct for all these arguments.

Reply:
Okay, I see. Yes, you're right. I'll do the same thing for both incremental and static and just pass down the options struct, picking out what is needed for the class.

Reply:
done